### DISCUSSION TITLE
Questions about gene scores

### DISCUSSION THREAD

**Discussion by Ping-lin14** [GreenleafLab/ArchR#613] [Questions / Documentation]
When I was performing Gene Score, I found a problem.
In my data, there is a cluster with high scores on both MS4A1 and CD4.
This is weird because I didn't see this on Signac. In fact, in Signac, there are very few cells with high MS4A1 scores and the distribution is very sporadic.
I think it may be a problem with the use of `addIterativeLSI `parameters. Do you have any suggestions for the use of `addIterativeLSI `parameters?
Currently I use iterations = 2, resolution = c(0.05, 0.1, 0.2, 0.3, 0.4, 0.5, 1), varFeatures = 100000.

**Comment by rcorces** [comment]
ArchR and Signac calculate gene scores in different ways. We cannot guarantee that ArchR's gene score is "right" and Signac's is "wrong" for each independent gene but our pretty extensive testing shows that ArchR's default model outperforms more basic models (Fig 2 and Extended Data Figs 6-7 in the manuscript - https://www.nature.com/articles/s41588-021-00790-6).

While your exact issue might be a difference in clustering, it seems likely that it is a difference in how the scores are calculated.

From the Signac webpage:

> Here we will use a simple approach of summing the fragments intersecting the gene body and promoter region

ArchR allows you to designate the parameters of the gene score model that you would like to use. I dont know what Signac considers as the "promoter" region though.

I would suggest adjusting the parameters of `addGeneScoreMatrix()` before playing around with clustering. However, its impossible to know which model is "correct" without some orthogonal validation. So it is equally possible that Signac's prediction of gene activity is wrong (or both are wrong).

https://www.archrproject.com/reference/addGeneScoreMatrix.html

**Reply by Ping-lin14** [reply]
I tried `addGeneScoreMatrix()`, but it doesn't seem to have much effect on my data.

I have an idea, in addition to gene scores, combined with transcription factor footprinting, 
It should be able to determine ethnic identity.

**Reply by rcorces** [reply]
Ok. I'm not sure exactly how I can best provide additional help but let me know.
